flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/SinkWriterOperator.java

```diff
@@ -185,12 +219,36 @@ private void emitCommittables(Long checkpointId) throws IOException, Interrupted
         Collection<CommT> committables =
                 ((PrecommittingSinkWriter<?, CommT>) sinkWriter).prepareCommit();
         StreamingRuntimeContext runtimeContext = getRuntimeContext();
-        int indexOfThisSubtask = runtimeContext.getIndexOfThisSubtask();
+        final int indexOfThisSubtask = runtimeContext.getIndexOfThisSubtask();
+        final int numberOfParallelSubtasks = runtimeContext.getNumberOfParallelSubtasks();
+
+        // Emit only committable summary if there are legacy committables
+        if (!legacyCommittables.isEmpty()) {
+            emit(
```

Comment:
Might add `checkState(checkpointId > InitContext.INITIAL_CHECKPOINT_ID)`?

Reply:
I guess it could be that the pipeline is started from a savepoint that is `InitContext.INITIAL_CHECKPOINT_ID`. I think checking `checkpointId >= InitContext.INITIAL_CHECKPOINT_ID` makes more sense. WDYT?

Reply:
Since it is called from `prepareSnapshotPreBarrier`, thus if the job is restored from savepoint, the next checkpoint should start from at least the `restored checkpoint id + 1`, thus it should always larger than `InitContext.INITIAL_CHECKPOINT_ID` ?

The main issues of `checkpointId == InitContext.INITIAL_CHECKPOINT_ID` is that in this case there will be two summary messages of the same checkpoint id, which would cause exceptions in `CommitterOperator`.